### PR TITLE
Fix issue #19270 (popovers and modals)

### PR DESCRIPTION
**Pull request by cristiancota**
I changed the $zindex-popover var because I found an unexpected behavior, when a Popover is showed and a modal is activated later, the popover it's over the modal.
Fix issue #19270.

### CODE REVIEW

File: scss/_popover.scss [hunk 1]
```diff
@@ -100,6 +100,9 @@
       }
     }
   }
+  &.inModal {
+    z-index: 1060;
```
Comment (cvrebert) [inline scss/_popover.scss]: Create a new variable and use it here. Hardcoding is bad.

File: scss/_popover.scss [hunk 2]
```diff
@@ -100,6 +100,9 @@
       }
     }
   }
+  &.inModal {
```
Comment (cvrebert) [inline scss/_popover.scss]: Our CSS class naming convention is `dash-separated-words`, not `camelCase`.

I advise renaming this to `.popover-in-modal` and un-nesting it.

File: js/src/tooltip.js [hunk 3]
```diff
@@ -242,6 +242,18 @@ const Tooltip = (($) => {
     show() {
       let showEvent = $.Event(this.constructor.Event.SHOW)
 
+      let isInModal = false
+
+      $(this.element).parents().each((index, elem) => {
```
Comment (cvrebert) [inline js/src/tooltip.js]: I think this can be simplified to:
```js
let isInModal  = $(this.element).parents('.modal.in').length > 0
```

**Comment by cvrebert** [comment]
This needs a unit test. See https://github.com/twbs/bootstrap/tree/v4-dev/js/tests#how-does-bootstraps-test-suite-work for info on how to add one.

**Comment by cvrebert** [comment]
You should make similar changes to tooltips also.

**Comment by cvrebert** [comment]
Perhaps the "hide when the modal hides" feature could be handled via a capture-phase event listener for the modal's hidden event, which would hide the popover-tooltip in response. I'm not sure.

**Comment by cristiancota** [comment]
@cvrebert I've done the changes that you told me before, but I'm not sure how to write the test because involves a modal, what should I do?

**Comment by cvrebert** [comment]
Put the tooltip/popover markup inside a `<div class="modal">`. You shouldn't need any of the other inner modal markup.
See https://github.com/twbs/bootstrap/blob/v4-dev/js/tests/unit/modal.js for examples of showing/hiding the modal and dealing with the modal's events.

**Comment by mdo** [comment]
Any feedback from @bardiharborow or @Johann-S on. this option? From a design perspective, it makes sense to clear these out for folks, but I'm unsure if this is the best way to do it.

**Comment by bardiharborow** [comment]
I've pushed an alternate solution over at #21227, which I prefer more than using a class like this.

**Comment by Johann-S** [comment]
I prefer @bardiharborow 's solution because it's solve the problem without rising the z-index value

**Comment by mdo** [comment]
Closing for #21227, thanks!

**Pull request closed without merging**